File: javascript/hamming/hamming.js
export function compute(strand1, strand2) {
  if (strand1.length !== strand2.length) {
    throw new Error('strands must be of equal length');
  }

  let diff = 0;

  for (let i = 0; i < strand1.length; i++) {
    if (strand1[i] !== strand2[i]) {
      diff++;
    }
  }

  return diff;
}

File: go/hamming/hamming.go
package hamming

import "errors"

// Distance calculates the Hamming Distance between two DNA strands.
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("left and right strands must be of equal length")
	}

	diff := 0

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			diff++
		}
	}

	return diff, nil
}
